infrastructure/src/main/java/org/corfudb/infrastructure/log/FileSystemAgent.java

```diff
@@ -155,35 +163,43 @@ private void checkLimits() {
     }
 
     public static class PartitionAgent {
+
+        private static final int UPDATE_INTERVAL = 5;
+        private static final int NO_DELAY = 0;
+
         @Getter
         private PartitionAttribute partitionAttribute;
         private final Path logPartition;
         private final FileSystemConfig config;
+
+        private static final ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
+
         public PartitionAgent(FileSystemConfig config) {
-            this.config =  config;
+            this.config = config;
             logPartition = Paths.get(config.logDir.getRoot().toString(), config.logDir.subpath(0, 1).toString());
-            partitionAttribute = setPartitionAttribute();
-            //we need a scheduler
+            initializeScheduler();
+        }
+
+        private void initializeScheduler(){
+            scheduler.scheduleWithFixedDelay(
+                    this::setPartitionAttribute, NO_DELAY, UPDATE_INTERVAL, SECONDS
+            );
```

Comment:
I don't see why this is needed.

Can you please answer this question https://github.com/CorfuDB/CorfuDB/pull/3112#discussion_r768248608

Reply:
After discussing with @xnull , some of the reasons are -
1. To avoid any additional latency of generating the PartitionAttributes during Failure Detection. It will further get cascaded into other processes that depend on FD.
2. To avoid duplicate calculations by other components than FD, like metrics (to be added in the future). These will also put additional overhead on disk IO.